### ISSUE TITLE
Rule that forbids using http URLs in string literals

### ISSUE THREAD

**Issue by alexandear** [mgechev/revive#1449]
**Is your feature request related to a problem? Please describe.**

We want to enforce secure URL usage across our Go codebase. Specifically, we want to forbid using `http://` URLs in string literals to avoid accidental use of insecure connections. This is hard to enforce in code reviews and easy to miss.

**Describe the solution you'd like**

A new rule `forbid-http-url` that detects string literals starting with `http://` and reports a violation. Ideally, the rule should be configurable with an allowed list or allow disabling in specific cases using suppression comments.

**Describe alternatives you've considered**

Adding a grep-based pre-commit hook to scan for `http://`:

```sh
#!/bin/sh

if grep -r --exclude-dir=.git 'http://' .; then
  echo "Found http:// URLs. Use https:// instead."
  exit 1
fi
```

**Additional context**

This rule would help enforce security best practices (sometimes) and catch situations like:

- https://github.com/ardanlabs/gotour/pull/266
- https://github.com/ardanlabs/gotour/pull/259
- https://github.com/golangci/golangci-lint/pull/5106
- https://github.com/google/go-github/pull/3657
- https://go-review.googlesource.com/c/website/+/602835 (this for *.md, but the idea is the same)

**Comment by chavacava** [comment]
Should the rule also cope with websockets URLs (`ws` vs `wss`) ?

What do you think @alexandear @denisvmedia @ccoVeille ?

**Comment by chavacava** [comment]
Draft version of the rule produces ~85% of the failures on testing code (`*_test.go`)
Should the rule skip tests?

What do you think @alexandear @denisvmedia @ccoVeille ?

**Comment by chavacava** [comment]
What about `unsecure-url-protocol` (or `unsecure-url-scheme` ?) in place of `forbid-http-url` as rule name?

What do you think @alexandear @denisvmedia @ccoVeille ?

**Comment by chavacava** [comment]
Should skip URL on `localhost` (and `127.0.0.1`) ? (~33% of failures on non-test files)

What do you think @alexandear @denisvmedia @ccoVeille ?

**Comment by ccoVeille** [comment]
> Should the rule also cope with websockets URLs (`ws` vs `wss`) ?
> 
> What do you think @alexandear @denisvmedia @ccoVeille ?

Why not

**Comment by ccoVeille** [comment]
> What about `unsecure-url-protocol` (or `unsecure-url-scheme` ?) in place of `forbid-http-url` as rule name?
> 
> What do you think @alexandear @denisvmedia @ccoVeille ?

unsecure-url-scheme sounds good

**Comment by ccoVeille** [comment]
> Draft version of the rule produces ~85% of the failures on testing code (`*_test.go`)
> Should the rule skip tests?
> 
> What do you think @alexandear @denisvmedia @ccoVeille ?

Skipping tests is fine.

**Comment by ccoVeille** [comment]
> Should skip URL on `localhost` (and `127.0.0.1`) ? (~33% of failures on non-test files)
> 
> What do you think @alexandear @denisvmedia @ccoVeille ?

Let's skip them
